src/client/rpc.rs

```diff
@@ -371,15 +371,28 @@ mod tests {
     use predicates::prelude::*;
     use reth_rpc_layer::{AuthLayer, JwtAuthValidator};
     use std::net::SocketAddr;
+    use std::net::TcpListener;
     use std::result::Result;
     use std::str::FromStr;
 
     use super::*;
 
-    const AUTH_PORT: u32 = 8550;
     const AUTH_ADDR: &str = "0.0.0.0";
     const SECRET: &str = "f79ae8046bc11c9927afe911db7143c51a806c4a537cc08e0d37140b0192f430";
 
+    fn get_available_port() -> u16 {
+        loop {
+            let port: u16 = rand::random_range(1000..20000);
+            if port_is_available(port) {
+                return port;
+            }
+        }
+    }
+
+    fn port_is_available(port: u16) -> bool {
+        TcpListener::bind(("127.0.0.1", port)).is_ok()
+    }
```

Comment:
IIRC you should be able to do something like the following to get an available port

```rust
fn get_available_port() -> u16 {
    let listener = TcpListener::bind("127.0.0.1:0").expect("Failed to bind");
    listener.local_addr().unwrap().port()
}
```

Reply:
This is less reliable, since the OS will tend to choose ports that are next to each other. So if you call this function twice in a row you'll be more likely to get the same port each time.

Reply:
You could do this if you kept a global state of claimed ports.

Reply:
@0xKitsune I Added a ClaimedPorts hashset to prevent collisions from back to back calls.